inference-engine/src/mkldnn_plugin/mkldnn_serialize.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "mkldnn_serialize.h"
+
+#include <transformations/serialize.hpp>
+
+#include <pugixml.hpp>
+
+using namespace InferenceEngine;
+
+namespace MKLDNNPlugin {
+namespace {
+    std::string to_string(InferenceEngine::Layout layout) {
+        std::stringstream ss;
+        ss << layout;
+        return ss.str();
+    }
+
+    InferenceEngine::Layout layout_from_string(const std::string & name) {
+        static const std::unordered_map<std::string, InferenceEngine::Layout> layouts = {
+            { "ANY", InferenceEngine::Layout::ANY },
+            { "NCHW", InferenceEngine::Layout::NCHW },
+            { "NHWC", InferenceEngine::Layout::NHWC },
+            { "NCDHW", InferenceEngine::Layout::NCDHW },
+            { "NDHWC", InferenceEngine::Layout::NDHWC },
+            { "OIHW", InferenceEngine::Layout::OIHW },
+            { "C", InferenceEngine::Layout::C },
+            { "CHW", InferenceEngine::Layout::CHW },
+            { "HWC", InferenceEngine::Layout::HWC },
+            { "HW", InferenceEngine::Layout::HW },
+            { "NC", InferenceEngine::Layout::NC },
+            { "CN", InferenceEngine::Layout::CN },
+            { "BLOCKED", InferenceEngine::Layout::BLOCKED }
+        };
+        auto it = layouts.find(name);
+        if (it != layouts.end()) {
+            return it->second;
+        }
+        IE_THROW(NetworkNotRead) << "Unknown layout with name '" << name << "'";
+    }
+
+    template<typename T>
+    void setPrecisionsAndLayouts(
+        pugi::xml_object_range<pugi::xml_named_node_iterator> && nodes,
+        T && info) {
+        for (auto n : nodes) {
+            auto name_attr = n.attribute("name");
+            auto precision_attr = n.attribute("precision");
+            auto layout_attr = n.attribute("layout");
+
+            if (!name_attr
+                || !precision_attr
+                || !layout_attr) {
+                IE_THROW(NetworkNotRead) << "The inputs/outputs information is invalid.";
+            }
+
+            auto it = info.find(name_attr.value());
+            if (it == info.end()) {
+                IE_THROW(NetworkNotRead) << "The input/output with name '" << name_attr.value() << "' not found";
+            }
+
+            it->second->setPrecision(Precision::FromStr(precision_attr.value()));
+            it->second->setLayout(layout_from_string(layout_attr.value()));
+        }
+    }
+};  // namespace
+
+CNNNetworkSerializer::CNNNetworkSerializer(std::ostream & ostream, MKLDNNExtensionManager::Ptr extensionManager)
```

Comment:
Why cannot we reuse default serializer?

Reply:
If you are talking about CNNNetwork::serialize, then unfortunately no. This implementation doesn't serialize CNN Network input and output precisions. A think, this behavior is incorrect for CNN Network serializer but I can't change existing implementation due to backward compatibility. Someone could use this serialization, and I can break his application.